neutrino-custom/index.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+const { join } = require('path');
+const { ProvidePlugin } = require('webpack');
+const lint = require('./lint');
+const development = require('./development');
+const production = require('./production');
+const test = require('./test');
+const notTest = require('./not-test');
+const localWatch = require('./local-watch');
+const react = require('neutrino-preset-react');
+const HtmlPlugin = require('html-webpack-plugin');
+
+const HTML_MINIFY_OPTIONS = {
+    useShortDoctype: true,
+    keepClosingSlash: true,
+    collapseWhitespace: true,
+    preserveLineBreaks: true
+};
+
+const entries = ['index', 'admin', 'perf', 'logviewer', 'failureviewer', 'userguide'];
+
+module.exports = neutrino => {
+    const { source } = neutrino.options;
+    const { NODE_ENV, LOCAL_WATCH } = process.env;
+
+    neutrino.config
+        // Environment-specific configuration
+        .when(NODE_ENV === 'development' && !LOCAL_WATCH, () => neutrino.use(development))
+        .when(NODE_ENV === 'production' || LOCAL_WATCH || NODE_ENV === 'test', () => neutrino.use(production))
```

Comment:
To answer the first question, yes, I would skip the creating a build and just do the test build (and a lint run).

To answer your second question, I totally agree. I tried to make the configurations converge as much as possible here without doing anything breaking (mostly upgrade now, change later).